gcloud/search/document.py

```diff
@@ -0,0 +1,351 @@
+    :param number_value: the actual value.
+    """
+
+    value_type = 'number'
+
+    def __init__(self, number_value):
+        self.number_value = number_value
+
+
+class TimestampValue(object):
+    """TimestampValues hold individual datetime values for a given field
+    See:
+    https://cloud.google.com/search/reference/rest/google/cloudsearch/v1/FieldValue
+
+    :type timestamp_value: class:``datetime.datetime``
+    :param timestamp_value: the actual value.
+    """
+
+    value_type = 'timestamp'
+
+    def __init__(self, timestamp_value):
+        self.timestamp_value = timestamp_value
+
+
+class GeoValue(object):
+    """GeoValues hold individual latitude/longitude values for a given field
+    See:
+    https://cloud.google.com/search/reference/rest/google/cloudsearch/v1/FieldValue
+
+    :type geo_value: tuple, (float, float)
+    :param geo_value: latitude, longitude
+    """
+
+    value_type = 'geo'
+
+    def __init__(self, geo_value):
+        self.geo_value = geo_value
+
+
+class Field(object):
+    """Fields hold values for a given document
+
+    See:
+    https://cloud.google.com/search/reference/rest/google/cloudsearch/v1/FieldValueList
+
+    :type name: string
+    :param name: field name
+    """
+
+    def __init__(self, name):
+        self.name = name
+        self.values = []
+
+    def add_value(self, value, **kw):
+        """Add a value to the field.
+
+        Selects type of value instance based on type of ``value``.
+
+        :type value: string, integer, float, datetime, or tuple (float, float)
+        :param value: the field value to add.
+
+        :param kw: extra keyword arguments to be passed to the value instance
+                   constructor.
+
+        :raises: ValueError if unable to match the type of ``value``.
+        """
+        if isinstance(value, six.string_types):
+            self.values.append(StringValue(value, **kw))
+        elif isinstance(value, (six.integer_types, float)):
+            self.values.append(NumberValue(value, **kw))
+        elif isinstance(value, datetime.datetime):
+            self.values.append(TimestampValue(value, **kw))
+        elif isinstance(value, tuple):
+            self.values.append(GeoValue(value, **kw))
+        else:
+            raise ValueError("Couldn't determine value type: %s" % (value,))
+
+
+class Document(object):
+    """Documents hold values for search within indexes.
+
+    See:
+    https://cloud.google.com/search/reference/rest/v1/projects/indexes/documents
+
+    :type name: string
+    :param name: the name of the document
+
+    :type index: :class:`gcloud.search.index.Index`
+    :param index: the index to which the document belongs.
+
+    :type rank: positive integer
+    :param rank: the default rank for ordering the document in queries:  if
+                 not passed, assigned a timestamp-based value by the server.
+                 See the ``rank`` entry on the page above for details.
```

Comment:
I asked for a clarification because I though `rank` might be an enum value that encodes a ranking strategy.

From the "page above" docs it's clear it is literally the rank of the document among its peers. Can you try to re-word so that is clearer? (I would usually try to come up with a suggestion in these situations but am in a rush, sorry.)